### ISSUE TITLE
Cope with IDs that are also alt_id

### ISSUE THREAD

**Issue by kimrutherford** [pombase/canto#1029]
The disease ontology has this obsolete term:

    [Term]
    id: DOID:10827
    name: gender identity disorder in adolescent or adult
    is_obsolete: true

But that term ID is the alt_id of this term:

    [Term]
    id: DOID:1234
    name: gender identity disorder
    alt_id: DOID:10827

the loader fails because it tries to make a dbxref for "DOID:10827" twice.  We need to load the obsolete terms because we have to cope with terms that become obsolete after the term is used in an annotation.

@airvine

**Comment by ValWood** [comment]
What do we do for GO?  That must cause the same situation?

**Comment by kimrutherford** [comment]
I just checked go-basic.obo and there are no alt_ids that are also the id of an obsolete term.  Perhaps GO doesn't do that when they obsolete a term?

**Comment by ValWood** [comment]
We do use alt_ids in GO but probably only for term merges. To me it seems like bad practice to make an obsolete ID a synonym of something else?   

If its the same thing why obsolete it? I think here GO would use a consider or replace by tag leading to the new ID.

If you are going to make something an alternative ID, it would be a term merge not an obsoletion.....

**Comment by ValWood** [comment]
Kim are there many examples?
And which loader? canto or chado?

**Comment by kimrutherford** [comment]
Alistair Irvine reported that he couldn't load the disease ontology (from HumanDO.obo) into Canto.  Haven't seen the problem anywhere else.

**Comment by cmungall** [comment]
Report it here: 

https://github.com/DiseaseOntology/HumanDiseaseOntology

and tag me

**Comment by kimrutherford** [comment]
In the D.O. case it looks like two terms were obsoleted (DOID:10827 and DOID:13350) and a new term (DOID:1234) created that had the obsolete terms as alt_ids.

**Comment by kimrutherford** [comment]
Reported: https://github.com/DiseaseOntology/HumanDiseaseOntology/issues/106

**Comment by mah11** [comment]
just a quick note to confirm that GO has no IDs that are simultaneously "id" and "alt_id" (regardless of obsoleteness)

**Comment by airvine** [comment]
I think is issue is more generally to do with IDs that are simultaneously alt_ids, and actually has nothing to do with obsoletion.

In the original example the disease ontology had two terms with alt_ids, both of which also were also used as term IDs in terms which happen to have been made obsolete. But when I looked at an earlier version of the disease ontology I also find a third term with an alt_id which is the term ID of another term, but in this case the term is not obsolete. In the later version of the ontology, this alt_id has been removed.

This is more of a bug in the disease ontology than in Canto, so we could just leave it until they remove the offending alt_ids. On the other hand, maybe it is good to cope with these minor errors without dismissing the entire ontology.
